homeassistant/components/cloudflare/__init__.py

```diff
@@ -97,20 +93,50 @@ async def _async_update_cloudflare(
 ) -> None:
     _LOGGER.debug("Starting update for zone %s", dns_zone["name"])
 
-    records = await client.list_dns_records(zone_id=dns_zone["id"], type="A")
+    records = await client.list_dns_records(zone_id=dns_zone["id"])
     _LOGGER.debug("Records: %s", records)
 
     session = async_get_clientsession(hass, family=socket.AF_INET)
-    location_info = await async_detect_location_info(session)
-
-    if not location_info or not is_ipv4_address(location_info.ip):
-        raise HomeAssistantError("Could not get external IPv4 address")
+    session_ipv6 = async_get_clientsession(hass, family=socket.AF_INET6)
 
-    filtered_records = [
-        record
-        for record in records
-        if record["name"] in target_records and record["content"] != location_info.ip
-    ]
+    location_info = await async_detect_location_info(session)
+    location_info_v6 = await async_detect_location_info(session_ipv6)
+
+    filtered_records = []
+
+    if location_info and is_ipv4_address(location_info.ip):
+        _LOGGER.info("IPv4 address detected: %s", location_info.ip)
```

Comment:
I think this should be `debug` level otherwise this will clutter logs?
```suggestion
        _LOGGER.debug("IPv4 address detected: %s", location_info.ip)
```